src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1851,10 +1853,19 @@ void parser::parse_and_visit_switch(parse_visitor_base &v) {
         });
         this->skip();
       } else {
-        this->parse_and_visit_expression(
+        expression *ast = this->parse_expression(
             v, precedence{
                    .colon_type_annotation = allow_type_annotations::never,
                });
+
+        source_code_span expression_case_span = ast->span();
+        if (repeated_cases.contains(expression_case_span.string_view())) {
+          this->diag_reporter_->report(
+              diag_duplicated_cases_in_switch_statement{
+                  .duplicated_switch_case = expression_case_span});
+        }
+        repeated_cases.insert(expression_case_span.string_view());
```

Comment:
Perf: We shouldn't do two lookups. Instead, use `insert` and see if it failed:
```suggestion
        auto [inserted, _it] = repeated_cases.insert(expression_case_span.string_view());
        if (!inserted) {
          this->diag_reporter_->report(
              diag_duplicated_cases_in_switch_statement{
                  .duplicated_switch_case = expression_case_span});
        }
```